inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasUimaParser.java

```diff
@@ -229,81 +225,161 @@ else if (annotation instanceof Sentence) {
                                 // char[].
                                 featureValue = String.valueOf((Object) WebAnnoCasUtil
                                         .getFeature(annotation, feature.getName()));
-                                
                                 // Add the UI annotation.feature name to the index as an annotation.
                                 // Replace spaces with underscore in the UI name.
-                                
                                 MtasToken mtasAnnotationFeature = new MtasTokenString(mtasId++,
-                                        annotationUiName.replace(" ", "_") + "."
-                                                + feature.getUiName().replace(" ", "_")
-                                                + MtasToken.DELIMITER + featureValue,
-                                        beginToken);
+                                        getIndexedName(annotationUiName) + "."
+                                        + getIndexedName(feature.getUiName()) 
+                                        + MtasToken.DELIMITER + featureValue, beginToken);
                                 mtasAnnotationFeature.setOffset(annotation.getBegin(),
                                         annotation.getEnd());
                                 mtasAnnotationFeature.addPositionRange(beginToken, endToken);
                                 tokenCollection.add(mtasAnnotationFeature);
+                                // Returns KB IRI label after checking if the  
+                                // feature type is associated with KB and feature value is not null
+                                String labelStr = "";
+                                if (feature.getType().contains(IndexingConstants.KB) && featureValue != "null") {
```

Comment:
Never use `==` or `!=` for string comparisons - use `equals`.

Reply:
Oh . My mistake. I was implementing a different logic but changed it later on. Will fix this